server/src/domain/errors/mod.rs

```diff
@@ -0,0 +1,60 @@
+use axum::http::StatusCode;
+use axum::response::{IntoResponse, Response};
+use axum::Json;
+use serde::Serialize;
+use serde_json::json;
+use thiserror::Error;
+use utoipa::ToSchema;
+
+/// Переназначение Result для ответов хендлеров.
+pub type Result<T, E = AppError> = core::result::Result<T, E>;
+
+/// Ошибки сервиса.
+#[derive(Debug, Error, ToSchema)]
+pub enum AppError {
+    // #[schema(example = "Some error from enum 'Errors'")]
+    // счет не существует
+    #[error("account with name: '{0}' not exists")]
+    AccountExistsErr(String),
+    // транзакция со значением 0
+    #[error("forbid transaction with 0 or less")]
+    ZeroValueTransactionErr,
+    // на счете не хватает средств для проведения транзакция
+    #[error("account balance less than operation value")]
+    OverdraftErr,
+    // транзакция самому себе
+    #[error("forbid transaction to yourself")]
+    SelfTransactionErr,
+    // транзакция не существует
+    #[error("account: '{0}' has no transaction with id: '{1}'")]
+    TransactionExistsErr(String, String),
+    // транзакция не существует
+    #[error("empty database")]
+    EmptyDbErr,
+    // ошибка загрузки файла репликации backup.json
+    #[error("backup load file error")]
+    BackupLoadFileErr,
+    // // остальные
+    // #[error(transparent)]
+    // Other(#[from] anyhow::Error),
+}
+
+/// Имплементация для Axum Response.
+impl IntoResponse for AppError {
+    fn into_response(self) -> Response {
+        let (status, err_msg) = match self {
+            AppError::AccountExistsErr(_) => (StatusCode::BAD_REQUEST, self.to_string()),
+            AppError::ZeroValueTransactionErr => (StatusCode::BAD_REQUEST, self.to_string()),
+            AppError::OverdraftErr => (StatusCode::BAD_REQUEST, self.to_string()),
+            AppError::SelfTransactionErr => (StatusCode::BAD_REQUEST, self.to_string()),
+            AppError::TransactionExistsErr(_, _) => (StatusCode::BAD_REQUEST, self.to_string()),
+            AppError::EmptyDbErr => (StatusCode::BAD_REQUEST, self.to_string()),
+            AppError::BackupLoadFileErr => (StatusCode::BAD_REQUEST, self.to_string()),
```

Comment:
```suggestion
            AppError::AccountExistsErr(_) |
            AppError::ZeroValueTransactionErr |
            AppError::OverdraftErr |
            AppError::SelfTransactionErr |
            AppError::TransactionExistsErr(_, _) |
            AppError::EmptyDbErr |
            AppError::BackupLoadFileErr => (StatusCode::BAD_REQUEST, self.to_string()),
```

Либо так, чтобы в будущем при добавлении нового варианта тут выскакивала ошибка компиляции, пока не будет исправлена, либо просто сразу возвращать `(StatusCode::BAD_REQUEST, self.to_string())` если считаем, что это всегда норм

Reply:
Ага, понял, спс. Выглядит неплохо. Поменял пару ошибок на `NOT_FOUND`.